File: BE/consultation-service/src/main/java/app/entity/Consultation.java
package app.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "consultations")
public class Consultation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private Date date;

    private int doctorId;

    private int animalId;

    private int ownerId;

    private String diagnostic;

    private String treatment;

    private String recommendations;

    private int price;


}

File: FE/src/app/models/Consultation.ts
import {Animal} from "./Animal";
import {Owner} from "./Owner";

export class Consultation {
  constructor(
    public id?: number,
    public date?: Date,
    public doctorId?: number,
    public animalId?: number,
    public ownerId?: number,
    public animalCode? : string,
    public doctorLastName?: string,
    public diagnostic?: string,
    public treatment?: string,
    public recommendations?: string,
    public price?: number,
    public consultatedAnimal: Animal = new Animal(),
    public owner : Owner = new Owner(),
    public image?: File
  ) {
    }
}
